enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,983 @@
+by annotating the IngressController with the annotation called `ingress.operator.openshift.io/auto-delete-load-balancer` 
+and then changing the eipAllocations by using the following commands:
+
+Annotate ingress controller with `ingress.operator.openshift.io/auto-delete-load-balancer` with a blank value.
+```sh
+oc -n openshift-ingress-operator annotate ingresscontrollers/test ingress.operator.openshift.io/auto-delete-load-balancer=
+```
+
+Update the Ingress Controller with the following command:
+
+```sh
+% oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"providerParameters":{"type":"AWS","aws":{"type":"NLB","nlbParameters":{"eipAllocations":[]}}}}}}}'
+```
+
+After the changes are saved, the operator will delete and create the load balancer service automatically.
+```sh
+% oc get svc router-test -n openshift-ingress 
+NAME          TYPE           CLUSTER-IP       EXTERNAL-IP                                                                     PORT(S)                      AGE
+router-test   LoadBalancer   172.30.230.109   a95d070adbbf045c9b356ed9db8d3cda-89fbf4dfc638408e.elb.us-east-1.amazonaws.com   80:30453/TCP,443:32704/TCP   77s
+```
+
+If you remove all EIPs, the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` is removed from the service as well.
+
+
+#### Unmanaged Subnet Annotation Migration Workflow
+
+Migrating an unmanaged `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+service annotation to `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancerParameters.eipAllocations`
+after upgrading to 4.y doesn't require a cluster admin to delete the LoadBalancer-type
+service:
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+   is set on the LoadBalancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.y and the service annotation is not
+   changed or removed. For more details see [section](#Any change to the service annotation does not trigger the operator to manage the load balancer service).  
+3. After upgrading, the IngressController will emit a `LoadBalancerProgressing` condition
+   with `Status: True` because the spec's `EIPAllocations` (an empty slice `[]`) does not equal
+   the current annotation.
+4. In this case, the cluster admin must directly update
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancerParameters.eipAllocations` to the
+   current value of the `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` service.
+5. The Ingress Operator will resolve the `LoadBalancerProgressing` condition back to
+   `Status: False` as long as `EIPAllocations` is equivalent to `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
```

Comment:
[Reposting](https://github.com/openshift/enhancements/pull/1593#discussion_r1631514651) for visibility here:

For installer-created-VPCs, how does a cluster admin accurately predict how many EIPAllocations to bring so their cluster doesn't fail to install? Maybe they don't know what public subnets will be created by the openshift-installer. 

I think it would be good to have a workflow for this. To illustrate further, the workflow could start like:

1. Cluster admin wants to use EIPs, but isn't bringing their own VPC, so they need to know exactly how many EIPs to create.
2. Cluster admin should review https://aws.amazon.com/about-aws/global-infrastructure/regions_az/ to find out how many AZs are in the region they are installing into.
3. Cluster admin should use the same number of EIPs as AZs in their region.

Let me know if that makes sense, or if anyone has a reason why this might be redundant. Cluster admins probably know how many AZs they are installing into, but I suppose a workflow like this would provide a concrete example of how this would happen if they don't.

Reply:
This workflow is very good. Thanks!

Reply:
@miheer did this get added?